### PR TITLE
[Merged by Bors] - feat(Logic/Equiv/TransferInstance): shrinking types preserves algebraic structures

### PR DESCRIPTION
**Pull request by jjaassoonn**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Logic/Equiv/TransferInstance.lean [hunk 1]
```diff
@@ -55,6 +56,10 @@ theorem one_def [One β] :
 #align equiv.one_def Equiv.one_def
 #align equiv.zero_def Equiv.zero_def
 
+@[to_additive]
+noncomputable instance [UnivLE.{u, v}] [One α] : One (Shrink.{v} α) :=
```
Comment (alreadydone) [inline Mathlib/Logic/Equiv/TransferInstance.lean:60]: I don't think you even need to use UnivLE in this file; `Small.{v} α` should work for this declaration, for example.
Maybe you only need to import Mathlib.Logic.Small.Defs (to reviewers: I think it's a pretty light import, and I think it's reasonable to add these instances in this file.)

File: Mathlib/Logic/Equiv/TransferInstance.lean [hunk 2]
```diff
@@ -575,6 +732,12 @@ def algEquiv (e : α ≃ β) [Semiring β] [Algebra R β] : by
         rfl }
 #align equiv.alg_equiv Equiv.algEquiv
 
+variable (α) in
+/-- Shrink `α` to a smaller universe algebra multiplication. -/
```
Comment (alreadydone) [inline Mathlib/Logic/Equiv/TransferInstance.lean:736]:
```suggestion
/-- Shrink `α` to a smaller universe preserves algebra structure. -/
```

File: Mathlib/Logic/Equiv/TransferInstance.lean [hunk 3]
```diff
@@ -295,6 +369,11 @@ protected def nonUnitalNonAssocSemiring [NonUnitalNonAssocSemiring β] :
   apply e.injective.nonUnitalNonAssocSemiring _ <;> intros <;> exact e.apply_symm_apply _
 #align equiv.non_unital_non_assoc_semiring Equiv.nonUnitalNonAssocSemiring
 
+
```
Comment (alreadydone) [inline Mathlib/Logic/Equiv/TransferInstance.lean:372]:
```suggestion
```

File: Mathlib/Logic/Equiv/TransferInstance.lean [hunk 4]
```diff
@@ -284,6 +353,11 @@ protected def commGroup [CommGroup β] : CommGroup α := by
 #align equiv.comm_group Equiv.commGroup
 #align equiv.add_comm_group Equiv.addCommGroup
 
+
```
Comment (alreadydone) [inline Mathlib/Logic/Equiv/TransferInstance.lean:356]:
```suggestion
```

File: Mathlib/Logic/Equiv/TransferInstance.lean [hunk 5]
```diff
@@ -258,6 +317,11 @@ protected def commMonoid [CommMonoid β] : CommMonoid α := by
 #align equiv.comm_monoid Equiv.commMonoid
 #align equiv.add_comm_monoid Equiv.addCommMonoid
 
+
```
Comment (alreadydone) [inline Mathlib/Logic/Equiv/TransferInstance.lean:320]:
```suggestion
```

File: Mathlib/Logic/Equiv/TransferInstance.lean [hunk 6]
```diff
@@ -271,6 +335,11 @@ protected def group [Group β] : Group α := by
 #align equiv.group Equiv.group
 #align equiv.add_group Equiv.addGroup
 
+
```
Comment (alreadydone) [inline Mathlib/Logic/Equiv/TransferInstance.lean:338]:
```suggestion
```

File: Mathlib/Logic/Equiv/TransferInstance.lean [hunk 7]
```diff
@@ -426,23 +543,36 @@ protected def commRing [CommRing β] : CommRing α := by
...
+noncomputable instance [Small.{v} α] [Nontrivial α] : Nontrivial (Shrink.{v} α) :=
+  (equivShrink α).symm.nontrivial
+
 /-- Transfer `IsDomain` across an `Equiv` -/
 @[reducible]
 protected theorem isDomain [Ring α] [Ring β] [IsDomain β] (e : α ≃+* β) : IsDomain α :=
   Function.Injective.isDomain e.toRingHom e.injective
 #align equiv.is_domain Equiv.isDomain
 
+
```
Comment (alreadydone) [inline Mathlib/Logic/Equiv/TransferInstance.lean:564]:
```suggestion
```

**Review by alreadydone — APPROVED** [review]
Thanks!

**Comment by alreadydone** [comment]
maintainer merge

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**